### PR TITLE
feat: disable MFE views for unsupported use-cases

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** Disable MFE views and fall back to old UI for unsupported use-cases.

JIRA: [AU-1507](https://2u-internal.atlassian.net/browse/AU-1507)

**What changed?**

Currently team assignments and ORAs with reordered assessment steps are unsupported in the MFE views. This adds a block of logic, `_mfe_views_supported`, to check, in addition to whether the feature is enabled for the course, whether we should use MFE views.

**Testing Instructions**

- Unit tests
- `learner_data` endpoint exercises

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

File: openassessment/xblock/openassessmentblock.py [hunk 1]
```diff
@@ -615,6 +617,34 @@ def student_view(self, context=None):  # pylint: disable=unused-argument
...
+        """
+
+        # Team assessments are currently unsupported
+        if self.is_team_assignment():
+            return False
+
+        # Assessment step reordering is currently unsupported
+        default_assessment_order = copy.copy(self.VALID_ASSESSMENT_TYPES)
+        current_assessment_order = self.assessment_steps
+        is_default_order = all(a <= b for a, b in zip(current_assessment_order, default_assessment_order[1:]))
```
Comment (jansenk) [inline openassessment/xblock/openassessmentblock.py:641]: What are the values of a and b here? less than or equal?

**Review by jansenk — APPROVED** [review]

File: openassessment/xblock/openassessmentblock.py [hunk 2]
```diff
@@ -615,6 +617,44 @@ def student_view(self, context=None):  # pylint: disable=unused-argument
...
+            step_index = default_assessment_order.index(assessment_step)
+
+            if step_index < last_step_index:
+                is_default_order = False
+                break
+            last_step_index = step_index
+
+        # Assessment step reordering is currently unsupported
+        if not is_default_order:
+            return False
```
Comment (jansenk) [inline openassessment/xblock/openassessmentblock.py:654]: Pull all this out into its own function, is_default_step_order. It will make mfe_views_supported more readable and it will avoid the "break and also check flag" thing, you'll just be able to return
